File: projects/01-promo-map/flutter_app/lib/data/models/review.dart
class Review {
  final int id;
  final int userId;
  final String userName;
  final int storeId;
  final int rating;
  final String content;
  final String createdAt;

  const Review({
    required this.id,
    required this.userId,
    required this.userName,
    required this.storeId,
    required this.rating,
    this.content = '',
    required this.createdAt,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['id'] as int,
      userId: json['user_id'] as int,
      userName: json['user_name'] as String,
      storeId: json['store_id'] as int,
      rating: json['rating'] as int,
      content: (json['content'] as String?) ?? '',
      createdAt: json['created_at'] as String,
    );
  }
}

File: projects/01-promo-map/backend/schemas/review.py
"""리뷰 스키마"""
from pydantic import BaseModel, field_validator
from typing import Optional

__all__ = ["ReviewCreateRequest", "ReviewResponse"]


class ReviewCreateRequest(BaseModel):
    store_id: int
    rating: int
    content: str = ""

    @field_validator("rating")
    @classmethod
    def validate_rating(cls, v):
        if v < 1 or v > 5:
            raise ValueError("Rating must be between 1 and 5")
        return v


class ReviewResponse(BaseModel):
    id: int
    user_id: int
    user_name: str
    store_id: int
    rating: int
    content: str
    created_at: str

    class Config:
        from_attributes = True
